cubi_tk/snappy/itransfer_common.py

```diff
@@ -431,6 +452,48 @@ def get_landing_zone_by_uuid(self, lz_uuid):
         )
         return lz.irods_path
 
+    def get_landing_zone_uuid_by_path(self, lz_irods_path, project_uuid, assay_uuid=None):
+        """
+        :param lz_path: Landing zone path.
+        :type lz_path: str
+
+        :param project_uuid: Project UUID.
+        :type project_uuid: str
+
+        :param assay_uuid: Assay UUID (optional).
+        :type assay_uuid: str
+
+        :return: Returns LZ UUID.
+        """
+        from sodar_cli.api import landingzone
+
+        # List existing lzs
+        existing_lzs = sorted(
+            landingzone.list_(
+                sodar_url=self.args.sodar_url,
+                sodar_api_token=self.args.sodar_api_token,
+                project_uuid=project_uuid,
+            ),
+            key=lambda x: x.date_modified,
+            reverse=True,
+        )
+
+        # Filter for assay
+        if assay_uuid:
+            existing_lzs = list(filter(lambda x: x.assay == assay_uuid, existing_lzs))
+
+        matching_lzs = list(filter(lambda x: x.irods_path == lz_irods_path, existing_lzs))
+        if matching_lzs:
+            lz_uuid = matching_lzs[0].sodar_uuid
+        else:
+            msg = "Could not find LZ with given path. Please review input: {0}".format(
```

Comment:
this doesn't seem to work. I tried uploading to a landing zone that I had previously deleted, and there was no error

Reply:
To prevent this a check for LZ status should be implemented. This is how I did it: https://github.com/bihealth/cubi-tk/blob/25baf0b2f21ae5125160314847f56910813408b5/cubi_tk/sodar/ingest.py#L126-L144

Reply:
There's no need to query the API again at that point - the LZ objects have a 'status' attribute. I've added a check that this needs to indicate an active LZ

Reply:
LT status can also be FAILED if a previous verification step failed. This LZ it's technically still active and can receive new files. In fact, a missing md5 file might be the reason verification failed. So you need to check for both status values.

The API does actually return the correct attribute "is_locked" as well, but the sodar_cli library does not know about this. There is a PR open for review about that.